pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -2734,6 +2735,121 @@ spec:
 	}
 }
 
+func TestStepActionRef(t *testing.T) {
+	taskRun := parse.MustParseV1TaskRun(t, `
+metadata:
+  name: test-taskrun-referencing-step-action
+  namespace: foo
+spec:
+  taskSpec:
+    steps:
+      - ref:
+          name: stepAction
+        name: step1
+        workingDir: /foo
+      - ref:
+          name: stepAction2
+        name: step2
+      - name: inlined-step
+        image: "inlined-image"
+`)
+	stepAction := parse.MustParseV1alpha1StepAction(t, `
+metadata:
+  name: stepAction
+  namespace: foo
+spec:
+  image: myImage
+  command: ["ls"]
+`)
+	stepAction2 := parse.MustParseV1alpha1StepAction(t, `
+metadata:
+  name: stepAction2
+  namespace: foo
+spec:
+  image: myImage
+  script: "echo hi"
+`)
+	d := test.Data{
+		TaskRuns:    []*v1.TaskRun{taskRun},
+		StepActions: []*v1alpha1.StepAction{stepAction, stepAction2},
+		ConfigMaps: []*corev1.ConfigMap{
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName(), Namespace: system.Namespace()},
+				Data: map[string]string{
+					"enable-step-actions": "true",
+				},
+			},
+		},
+	}
+	testAssets, cancel := getTaskRunController(t, d)
+	defer cancel()
+	createServiceAccount(t, testAssets, "default", taskRun.Namespace)
+	c := testAssets.Controller
+	if err := c.Reconciler.Reconcile(testAssets.Ctx, getRunName(taskRun)); err == nil {
+		t.Fatalf("Could not reconcile the taskrun: %v", err)
+	}
+	getTaskRun, _ := testAssets.Clients.Pipeline.TektonV1().TaskRuns(taskRun.Namespace).Get(testAssets.Ctx, taskRun.Name, metav1.GetOptions{})
+	got := getTaskRun.Status.TaskSpec.Steps
+	want := []v1.Step{{
+		Image:      "myImage",
+		Command:    []string{"ls"},
+		Name:       "step1",
+		WorkingDir: "/foo",
+	}, {
+		Image:  "myImage",
+		Script: "echo hi",
+		Name:   "step2",
+	}, {
+		Image: "inlined-image",
+		Name:  "inlined-step",
+	}}
+	if c := cmp.Diff(want, got); c != "" {
+		t.Errorf("TestStepActionRef errored with: %s", diff.PrintWantGot(c))
+	}
+}
+
+func TestStepActionRef_Error(t *testing.T) {
+	taskRun := parse.MustParseV1TaskRun(t, `
+metadata:
+  name: test-taskrun-referencing-step-action
+  namespace: foo
+spec:
+  taskSpec:
+    steps:
+      - name: missing-step-action
+        ref:
+          name: noStepAction
+`)
+	d := test.Data{
+		TaskRuns: []*v1.TaskRun{taskRun},
+		ConfigMaps: []*corev1.ConfigMap{
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName(), Namespace: system.Namespace()},
+				Data: map[string]string{
+					"enable-step-actions": "true",
+				},
+			},
+		},
+	}
+	testAssets, cancel := getTaskRunController(t, d)
+	defer cancel()
+	createServiceAccount(t, testAssets, "default", taskRun.Namespace)
+	c := testAssets.Controller
+	c.Reconciler.Reconcile(testAssets.Ctx, getRunName(taskRun))
```

Comment:
Is there an err returned here? If so can we check that err as well?

Reply:
fixed! Thanks!